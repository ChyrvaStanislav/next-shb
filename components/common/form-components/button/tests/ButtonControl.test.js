import React from 'react';
import { Formik, Form } from 'formik';
import { render } from '@testing-library/react';
import ButtonControll from '../ButtonControl';

const Component = props => (
  <Formik
    initialValues={{
      commercialUsage: false,
    }}
    isInitialValid={true}
  >
    <ButtonControll {...props} />
  </Formik>
);

const onClickCancel = jest.fn();

describe('Link renders', () => {
  it('should ButtonControl renders', () => {
    const { asFragment } = render(<Component />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('should Preloader renders', () => {
    const { asFragment } = render(<Component loading={true} />);
    expect(asFragment).toMatchSnapshot();
  });
});

describe('Disable and hide submit button', () => {
  it('should submit button hidden when isSubmitHidden', () => {
    const { queryByText } = render(<Component isSubmitHidden={true} />);
    const buttonControlElem = queryByText('Send');
    expect(buttonControlElem).not.toBeInTheDocument();
  });

  it('should submit button disabled when toShowRadioButtons & commercialUsage', () => {
    const ComponentWithForm = () => (
      <Formik initialValues={{
        commercialUsage: false,
      }}
      >
        <Form>
          <ButtonControll toShowRadioButtons={true} />
        </Form>
      </Formik>
    );
    const { container } = render(<ComponentWithForm />);
    const submitButtonByTagName = container.querySelector('.disabled');
    expect(submitButtonByTagName).toBeInTheDocument();
  });

  it('should submit button disabled when loading', () => {
    const { container } = render(<Component loading={true} />);
    const submitButtonByTagName = container.querySelector('.disabled');
    expect(submitButtonByTagName).toBeInTheDocument();
  });

  it('should submit button disabled when form isValid = false', () => {
    const ComponentWithForm = () => (
      <Formik
        isInitialValid={false}
      >
        <Form>
          <ButtonControll />
        </Form>
      </Formik>
    );
    const { container } = render(<ComponentWithForm />);
    const submitButtonByTagName = container.querySelector('.disabled');
    expect(submitButtonByTagName).toBeInTheDocument();
  });
});


describe('Cancel onClick', () => {
  it('should cancel onClick works', () => {
    const { queryByText } = render(<Component onClickCancel={onClickCancel} />);
    const cancelButton = queryByText('Cancel');
    cancelButton.click();
    expect(onClickCancel).toBeCalled();
  });
});
