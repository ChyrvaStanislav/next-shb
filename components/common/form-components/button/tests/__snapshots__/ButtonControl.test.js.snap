// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders should ButtonControl renders 1`] = `
<DocumentFragment>
  <section
    class="componentContainer"
  >
    <div
      class="submitWrapper"
    >
      <button
        class="button primary medium send btn SendBtn"
        type="submit"
      >
        Send
      </button>
    </div>
    <button
      class="button outline medium cancel btn"
      type="button"
    >
      Cancel
    </button>
  </section>
</DocumentFragment>
`;

exports[`Link renders should Preloader renders 1`] = `[Function]`;
